subxt/src/client/light_client/builder.rs

```diff
@@ -147,10 +162,68 @@ impl<T: Config> LightClientBuilder<T> {
             user_data: (),
         };
 
-        let rpc_client = RpcClient::new(LightClientRpc::new(config)?);
-        let online_client = OnlineClient::<T>::from_rpc_client(rpc_client).await?;
-        Ok(LightClient(online_client))
+        let raw_rpc = LightClientRpc::new(config)?;
+        build_client_from_rpc(raw_rpc).await
+    }
+}
+
+/// Raw builder for [`RawLightClient`].
+pub struct RawLightClientBuilder {
+    chains: Vec<subxt_lightclient::AddedChain>,
+}
+
+impl Default for RawLightClientBuilder {
+    fn default() -> Self {
+        Self { chains: Vec::new() }
+    }
+}
+
+impl RawLightClientBuilder {
+    /// Create a new [`RawLightClientBuilder`].
+    pub fn new() -> RawLightClientBuilder {
+        RawLightClientBuilder::default()
     }
+
+    /// Adds a new chain to the list of chains synchronized by the light client.
+    pub fn add_chain(
+        mut self,
+        chain_id: subxt_lightclient::ChainId,
+        rpc_responses: subxt_lightclient::JsonRpcResponses,
+    ) -> Self {
+        self.chains.push(subxt_lightclient::AddedChain {
+            chain_id,
+            rpc_responses,
+        });
+        self
+    }
+
+    /// Construct a [`RawLightClient`] from a raw smoldot instance.
```

Comment:
Just to check, is this a `smoldot` instance or specifically just a `smoldot_lightclient::Client` instance?

Reply:
The `subxt_lightclient::Client` in this case is a re-export of `smoldot_light::Client`. Will modify the docs to say `client` instead of instance here and everywhere else, thanks!

Reply:
See my other comment re renaming exports a bit :)